### PR TITLE
Add support for new properties and params on Invoice and Subscription

### PR DESCRIPTION
**Pull request by remi-stripe**
* Add `transfer_data[destination]` to both
* Add `application_fee_amount` to Invoice

cc @stripe/api-libraries 

NOT asking for review yet as I want to double check that we don't need `on_behalf_of` for now.

### CODE REVIEW

**Comment by remi-stripe** [comment]
This is now ready for review. It's not yet public but it's something we want to support in the libraries for a while before we ship it publicly.

r? @mickjermsurawong-stripe

File: src/main/java/com/stripe/model/Invoice.java [hunk 1]
```diff
@@ -450,4 +459,8 @@ public Invoice voidInvoice(Map<String, Object> params, RequestOptions options)
     List<String> lineItemIds;
     Long usageGte;
   }
+
+  public static class TransferData extends StripeObject {
+    String destination;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Invoice.java]: whoops i think we might need these magic to make the fields accessible.. 
for setter just for consistency. Same for the `Subscription#TransferData` too.

```
  @Getter
  @Setter
  @EqualsAndHashCode(callSuper = false)
```

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Nice catch! I went to add a test but since it's gated stripe-mock doesn't support it yet so I put a TODO for now. PTAL

**Comment by mickjermsurawong-stripe** [comment]
LGTM

**Comment by remi-stripe** [comment]
I'm going to go ahead and merge + release even though the approved label did not unblock.

**Pull request merged**